### PR TITLE
docs: renames user guides to tasks

### LINKED ISSUE #3098: Rename "user guides" to "tasks"
*Description*:

In the [envoy gateway docs](https://gateway.envoyproxy.io/v1.0.0/), one of the main sections is "User Guides."

The title "User Guides" doesn't accurately describe its contents, which are one-off tasks.
Can we rename "User Guides" to "Tasks"?
That's what Kubernetes and Istio call theirs.

[optional *Relevant Links*:]
- https://istio.io/latest/docs/
- https://kubernetes.io/docs/home/

### PR DESCRIPTION
**Pull request by eitansuez**
Fixes #3098

### CODE REVIEW

**Comment by eitansuez** [comment]
this commit is a starting point.  there are broken links that i'll try to chase down.  probably due to renaming the `user/` folder to `tasks/`.  i tried to revise the links in latest/ but not retroactively, that could be the issue.  feel free to reject this PR if you think this goes down the wrong path.

**Review by zirain — APPROVED** [review]

File: site/content/en/latest/contributions/DEVELOP.md [hunk 1]
```diff
@@ -140,24 +140,24 @@ There are many other endpoints on the [Envoy admin interface][] that may be help
...
 
-[Quickstart]: https://github.com/envoyproxy/gateway/blob/main/docs/latest/user/quickstart.md
+[Quickstart]: https://github.com/envoyproxy/gateway/blob/main/docs/latest/tasks/quickstart.md
 [make]: https://www.gnu.org/software/make/
 [Github Actions]: https://docs.github.com/en/actions
 [workflows]: https://github.com/envoyproxy/gateway/tree/main/.github/workflows
 [Kind]: https://kind.sigs.k8s.io/
 [conformance homepage]: https://gateway-api.sigs.k8s.io/concepts/conformance/
-[directly exposing]: https://kind.sigs.k8s.io/docs/user/loadbalancer/
+[directly exposing]: https://kind.sigs.k8s.io/docs/tasks/loadbalancer/
```
Comment (Xunzhuo) [inline site/content/en/latest/contributions/DEVELOP.md:154]: This is 404

Comment (eitansuez) [reply site/content/en/latest/contributions/DEVELOP.md]: actually that links is to kubernetes kind - i think it's fine.

**Comment by eitansuez** [comment]
some markdowns reference published docs, so they can't yet reference the tasks/ folder - in order to pass the link check.  after this is merged, they'll need to be updated.

File: site/content/en/latest/contributions/RELEASING.md [hunk 2]
```diff
@@ -6,7 +6,10 @@ description: "This section tells the release process of Envoy Gateway."
 This document guides maintainers through the process of creating an Envoy Gateway release.
 
 - [Release Candidate](#release-candidate)
+  - [Prerequisites](#prerequisites)
```
Comment (eitansuez) [inline site/content/en/latest/contributions/RELEASING.md:9]: this is wrong.  my vs code is inserting these automatically on save.  will work on a commit to fix/revert this.

**Comment by eitansuez** [comment]
i suspect re-running the build will make it pass.

**Comment by eitansuez** [comment]
This pr is ready to merge

**Review by Xunzhuo — APPROVED** [review]

**Comment by arkodg** [comment]
hey @eitansuez can we also include this in v1.0 ?

**Pull request merged**